docs/Controllers.md

```diff
@@ -0,0 +1,87 @@
+
+If a Controller receives a HTTP 500 response code from the API, a failure has occurred.
+The Controller SHOULD display the content of the response�s error field to the user if possible, and indicate that the Resource may be in an unknown state.
+The Controller SHOULD also refresh the endpoints of the relevant Resources to ensure the Controller is accurately reflecting the current state of the API.
+
+## Registry Service Discovery	
+
+In order to locate the Registry, the Controller SHOULD support all of the following: 
+* Unicast DNS Service Discovery (DNS-SD).
+* mDNS.
+* Direct configuration with the location of a preferred Registry.
+
+The Controller SHOULD offer unicast DNS-SD as the default mechanism. 
+
+Controllers SHOULD observe and interpret all of the TXT records returned with the DNS Service Discovery responses according to the requirements for Query API Clients specified in the [API Paths section of the APIs document](APIs.md#api-paths) in this specification.
+
+## Query API
+The Controller MUST be capable of using the Registry's IS-04 Query API to discover any registered resource, including Node, Device,  Source, Flow, Sender, and Receiver,
+as described in the [APIs section](README.md#apis) of this specification.
+
+The Controller MUST use the Registry�s IS-04 Query API either via the REST API or by requesting WebSocket subscriptions.
+
+When using the Query API, basic queries SHOULD be used (and advanced query language where available) to cut down on the volume of resources returned to the Controller.	
+
+Controllers SHOULD adhere to the version downgrade requirements for Query API Clients specified in the [Requirements for Query API Clients section of the Upgrade Path document](Upgrade%20Path.md#requirements-for-query-api-clients) in this specification.
+
+## Pagination
+In large systems, API resources rely upon pagination in order to return high volumes of data.
+This can be both hard to keep track of for a client, and require a large number of requests in order to scan the entire data set.
+
+For this reason it is recommended to use the RESTful QUERY API for debug and development purposes only.
+
+If using the RESTful Query API rather than WebSockets, Pagination requirements MUST be implemented as specified in the [Pagination section of the APIs: Query Parameters document](APIs%20-%20Query%20Parameters.md#pagination) in this specification.
+	
+## WebSockets & Subscriptions	
+Where a WebSocket or other subscription based mechanism is provided for Controller usage, it is strongly recommended that Controllers make use of this and do not use the API resources directly.
+
+If a WebSocket connection fails, then an attempt to reconnect to the WebSocket SHOULD NOT be attempted. Instead, a new subscription SHOULD be created with this API or a different one if required. If all available APIs return errors, an exponential backoff algorithm SHOULD be used when retrying until a success code is returned.
```

Comment:
Why not attempt a reconnection to the same WebSocket??

Reply:
That justification is plain wrong as far as I'm aware. Reconnecting to a WebSocket results in all resources that match the `params` being sent again!

Reply:
Assuming you mean that the advice on the Wiki is plain wrong (rather than my paraphrasing) I feel we might need the originator of the Wiki advice to make their case before we remove it from the PR (and by extension, the Wiki?). @peterbrightwell do you know where this advice originated?

Reply:
This will have come from @andrewbonney some time ago.  I suggest a slack discussion with @garethsb to "sync" everyone's position :)

Reply:
Added related issue https://github.com/AMWA-TV/nmos-testing/issues/664.

What are we going to do here then... Just delete this paragraph, or is there something else we can say?

Reply:
https://github.com/AMWA-TV/is-04/blob/v1.2.x/docs/4.2.%20Behaviour%20-%20Querying.md#websocket-protocol says:

> Using data Grains, the websocket protocol provides a client with the current state of the contents of the registry at the time of _subscription (my emphasis)_. It subsequently notifies clients of any changes.

I'm 99% certain that was intended to mean "at the time of _connection_" but I'd like @andrewbonney's opinion...

Reply:
Ah - just noticed I was on the v1.2.x docs there... sure enough the language is corrected in v1.3.x - https://github.com/AMWA-TV/is-04/blob/v1.3.x/docs/Behaviour%20-%20Querying.md#websocket-messages says:

> The server provides a client with the current state of the contents of the registry at the time of their WebSocket connection, using messages which are known as data Grains. It subsequently notifies clients of any changes via additional data Grain messages.

Reply:
Ah - just noticed I was on the v1.2.x docs there... sure enough the language is corrected in v1.3.x - https://github.com/AMWA-TV/is-04/blob/v1.3.x/docs/Behaviour%20-%20Querying.md#websocket-messages says:

> The server provides a client with the current state of the contents of the registry at the time of their WebSocket connection, using messages which are known as data Grains. It subsequently notifies clients of any changes via additional data Grain messages.

Reply:
OK, is the conclusion here that we remove line 79?